prow/flagutil/github.go

```diff
@@ -102,7 +108,9 @@ func (o *GitHubOptions) GitHubClient(secretAgent *secret.Agent, dryRun bool) (cl
 
 // GitClient returns a Git client.
 func (o *GitHubOptions) GitClient(secretAgent *secret.Agent, dryRun bool) (client *git.Client, err error) {
-	client, err = git.NewClient()
+	// We already validated this during flag validation
```

Comment:
The common pattern we use in the rest of our config is to have a field in the options to store parsed values, and just access them.

Reply:
Sounds like overoptimization to me. Also, I am failing to find an example of this pattern in command line handling.
Could you please provide an example?

Reply:
Example in `ghproxy` options with a URL, and in the brancher, regex matcher, and trigger for a regex in Prow config. It's not over optimization. We should use the type system to enforce the flow of our program. We validate that we have a URL at the flag loading time, and we know for a fact that we do not have an invalid URL after that point. Having to re-parse it and ignore errors on parsing downstream is misleading.